cluster-api/cloud/google/machineactuator.go

```diff
@@ -285,7 +285,10 @@ func (gce *GCEClient) Delete(machine *clusterv1.Machine) error {
 
 	if instance == nil {
 		glog.Infof("Skipped deleting a VM that is already deleted.\n")
-		return nil
+		// Remove the finalizer
+		machine.ObjectMeta.Finalizers = util.Filter(machine.ObjectMeta.Finalizers, clusterv1.MachineFinalizer)
+		_, err = gce.machineClient.Update(machine)
+		return err
```

Comment:
Wouldn't it be better to fix this here: https://github.com/kubernetes/kube-deploy/blob/master/cluster-api/pkg/controller/machine/controller.go#L83

Reply:
I second this. That way, you only have the finalizer removal logic in one place instead of twice in this method.